src/cli/cmd/login/mod.rs

```diff
@@ -408,7 +408,25 @@ pub async fn upsert_user_nix_config(
     Ok(())
 }
 
-pub(crate) fn user_auth_token_path() -> Result<PathBuf, FhError> {
+pub(crate) async fn user_auth_token_read_path() -> Result<PathBuf, FhError> {
+    let write_path = user_auth_token_write_path();
+
+    if let Ok(ref path) = write_path {
+        if tokio::fs::metadata(&path).await.is_ok() {
+            return Ok(path.to_path_buf());
+        }
+    }
```

Comment:
super tiny nit: could we change this into a `match`? AFAICT, the only time we'd get to returning `write_path` below is if it's an error

It's not clear that this is the case at a glance though

Reply:
(This would have the added bonus of us not needing to borrow and then allocate and turn the `path` back into a pathbuf in the `Ok(_)` case)

Reply:
there are two errors here: one if we can't get the path via xdg, and one if the file does't exist. If the file doesn't exist, we may return `write_path` which will be Ok(Path). This makes it less easy to make it a match, but I could at a minimum explain this ... :)